libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1551,8 +1551,14 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 
 	if (!functionType->takesArbitraryParameters() && parameterTypes.size() != arguments.size())
 	{
+		string kindStr;
```

Comment:
Why not call it `kind`?

Alternatively have a `bool isStructConstructorCall` and have a ternary operator in `msg`.

Reply:
Yes, the boolean makes more sense, I hadn't given it much thought really.